src/main/java/Gluti/helpers/FileStorage.java

```diff
@@ -0,0 +1,136 @@
+package Gluti.helpers;
+
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import Gluti.utils.Task;
+import Gluti.utils.Todo;
+import Gluti.utils.Event;
+import Gluti.utils.Deadline;
+/**
+ * Provides a Storage interface that is shared among the program
+ * Loads tasks from file when program is ran
+ * and saves the tasks to file when terminated correctly
+ */
+
+public class FileStorage {
+    protected static final File DATA_FOLDER = new File("./src/main/data");
+    protected static final File DATA_FILE = new File("./src/main/data/Gluti.txt");
+    protected ArrayList<Task> tasklist = new ArrayList<>();
+
+    /**
+     * Initialises a FileStorage instance
+     * Check if there already exists a file, load tasks from file if exists
+     * Creates a new file if !exists
+     */
+    public FileStorage() {
+        checkIfExist();
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(DATA_FILE));
+            String line = reader.readLine();
+            while (line != null) {
+                tasklist.add(read(line));
+                line = reader.readLine();
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+//        System.out.println("Done mounting tasks");
+    }
+
+    /**
+     * Interprets the tasks loaded from file
+     * @param line the line read from File
+     * @return the Task interpreted from program
+     */
+
+    public Task read(String line) {
+        assert line != null : "Next line read should not be empty";
+        char tasktype = line.charAt(1);
```

Comment:
Should your variable name be camelCase instead of nocase?
```suggestion
        char taskType = line.charAt(1);
```